cl/sentinel/service/service.go

```diff
@@ -366,3 +350,45 @@ func (s *SentinelServer) GetPeersStatistics() map[string]*diagnostics.PeerStatis
 
 	return stats
 }
+
+func (s *SentinelServer) trackPeerStatistics(peerID string, inbound bool, msgType string, msgCap string, bytes int) {
+	if s.peerStatistics == nil {
+		s.peerStatistics = make(map[string]*diagnostics.PeerStatistics)
+	}
+
+	if _, exists := s.peerStatistics[peerID]; !exists {
+		s.peerStatistics[peerID] = &diagnostics.PeerStatistics{
+			CapBytesIn:   make(map[string]int),
+			CapBytesOut:  make(map[string]int),
+			TypeBytesIn:  make(map[string]int),
+			TypeBytesOut: make(map[string]int),
+		}
+	}
+
+	stats := s.peerStatistics[peerID]
+
+	if inbound {
+		stats.BytesIn += bytes
+		stats.CapBytesIn[msgCap] += bytes
+		stats.TypeBytesIn[msgType] += bytes
+	} else {
+		stats.BytesOut += bytes
+		stats.CapBytesOut[msgCap] += bytes
+		stats.TypeBytesOut[msgType] += bytes
+	}
+}
+
+func parseTopick(input string) (string, string) {
+	re := regexp.MustCompile(`^/([^/]+)/[^/]+/([^/]+)`)
```

Comment:
Typo should be Topic

Also for performance reasons do you really need a regexp here.  I think you should be using the golang strings lib and using split or cut.